include/wil/cppwinrt_authoring.h

```diff
@@ -0,0 +1,284 @@
+        typename Xaml_Data_PropertyChangedEventArgs = wil::details::Xaml_Data_PropertyChangedEventArgs>
+    struct single_threaded_notifying_property : single_threaded_rw_property<T>
+    {
+        using Type = T;
+        using base_type = single_threaded_rw_property<T>;
+        using base_type::operator();
+
+        template<typename Q> auto& operator()(Q&& q)
+        {
+            return *this = std::forward<Q>(q);
+        }
+
+        template<typename Q> auto& operator=(Q&& q)
+        {
+            if (q != this->operator()())
+            {
+                static_cast<base_type&>(*this) = std::forward<Q>(q);
+                if (auto strong = m_sender.get(); (m_npc != nullptr) && (strong != nullptr))
+                {
+                    (*m_npc)(strong, Xaml_Data_PropertyChangedEventArgs{ m_name });
+                }
+            }
+            return *this;
+        }
+
+        template<typename... TArgs>
+        single_threaded_notifying_property(
+            winrt::event<Xaml_Data_PropertyChangedEventHandler>* npc,
+            const winrt::Windows::Foundation::IInspectable& sender,
+            std::wstring_view name,
+            TArgs&&... args) :
+            single_threaded_rw_property<T>(std::forward<TArgs...>(args)...),
+            m_name(name),
+            m_npc(npc),
+            m_sender(sender)
+        {}
+
+        single_threaded_notifying_property(const single_threaded_notifying_property&) = default;
+        single_threaded_notifying_property(single_threaded_notifying_property&&) = default;
+        std::wstring_view Name() const noexcept { return m_name; }
+    private:
+        std::wstring_view m_name;
+        winrt::event<Xaml_Data_PropertyChangedEventHandler>* m_npc;
+        winrt::weak_ref<winrt::Windows::Foundation::IInspectable> m_sender;
+    };
+
+    /**
+    * @def WIL_NOTIFYING_PROPERTY
+    * @brief use this to stamp out a property that calls RaisePropertyChanged upon changing its value. This is a zero-storage alternative to wil::single_threaded_notifying_property<T>
+    * @details You can pass an initializer list for the initial property value in the variadic arguments to this macro.
+    */
+#define WIL_NOTIFYING_PROPERTY(type, name, ...)             \
+    type m_##name{__VA_ARGS__};                              \
+    auto name() const noexcept { return m_##name; }         \
+    auto& name(type value)                                  \
+    {                                                       \
+        if (m_##name != value)                              \
+        {                                                   \
+            m_##name = std::move(value);                    \
+            RaisePropertyChanged(L#name);                   \
+        }                                                   \
+        return *this;                                       \
+    }                                                       \

```

Comment:
```suggestion
#define WIL_NOTIFYING_PROPERTY(type, name, ...)             \
private:                                                    \
    type m_##name{__VA_ARGS__};                             \
public:                                                     \
    auto name() const noexcept { return m_##name; }         \
    auto& name(type value)                                  \
    {                                                       \
        if (m_##name != value)                              \
        {                                                   \
            m_##name = std::move(value);                    \
            RaisePropertyChanged(L"" #name);                \
        }                                                   \
        return *this;                                       \
    }                                                       \
```
Note that `L#name` isn't valid. We could also use `protected` over `private`, but this should be fine unless a legitimate need arises.

Reply:
L#x is valid. https://godbolt.org/z/73jWs4vv8

Reply:
[No it isn't](https://godbolt.org/z/r1797ar6b)

MSVC's preprocessor support is non-standard. There are flags to enable standards conformance, though

Reply:
Some more info: https://learn.microsoft.com/en-us/cpp/preprocessor/preprocessor-experimental-overview?view=msvc-170#lval

Reply:
huh, TIL. Fixed.